toolchain/sem_ir/formatter.cpp

```diff
@@ -821,54 +821,45 @@ auto Formatter::FormatPendingConstantValue(AddSpace space_where) -> void {
 }
 
 auto Formatter::FormatInstLhs(InstId inst_id, Inst inst) -> void {
-  switch (inst.kind().value_kind()) {
-    case InstValueKind::Typed:
+  switch (inst.kind()) {
+    case InstKind::ImplWitnessTable:
+    case InstKind::ImportCppDecl:
+    case InstKind::ImportDecl:
+    case InstKind::ImportRefUnloaded:
+      // Although these don't have a typed value, we still want to print the
+      // name.
```

Comment:
I wonder if we should add back the `InstValueKind::Untyped` that we used to have for this?

Reply:
Here's my read on this:

- `InstValueKind` is actually just whether there's a `type_id` member. Adding another entry to the enum would make the calculation more complex.
- Of the 3 places that mention `InstValueKind` right now, the other two actually only care about typed values.
- Whereas `ImportRefUnloaded` is an untyped value, I'd be wary of saying the same of `ImportDecl`.

An alternative might be to drop the enum. Thoughts on this `has_type` approach?

Reply:
Split off to #5495